### ISSUE TITLE
consent info is now returning an invalid value

### ISSUE THREAD

**Issue by MonkeyDoDev** [blaugold/iabtcf_consent_info#13]
Sorry, relative beginner developer here but hopefully i can provide the info you need.

My implementation had been working fine... However a couple of days ago my calls retrieve .currencyConsentInfo() started failing due to an out of range error. I guess something about the values the google UMP outputs must have changed? 

My Crashlytics report is below incase it helps. 

/flutter (17455): ----------------FIREBASE CRASHLYTICS----------------
I/flutter (17455): RangeError (index): Invalid value: Not in inclusive range 0..9: 10
I/flutter (17455): #0      _Array.[] (dart:core-patch/array.dart:10:36)
I/flutter (17455): #1      _parseDataUsagePurposeBinaryString.<anonymous closure> (package:iabtcf_consent_info/iabtcf_consent_info.dart:290:45)
I/flutter (17455): #2      MappedListIterable.elementAt (dart:_internal/iterable.dart:415:31)
I/flutter (17455): #3      ListIterator.moveNext (dart:_internal/iterable.dart:344:26)
I/flutter (17455): #4      new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
I/flutter (17455): #5      new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
I/flutter (17455): #6      new List.of (dart:core-patch/array_patch.dart:47:28)
I/flutter (17455): #7      ListIterable.toList (dart:_internal/iterable.dart:214:7)
I/flutter (17455): #8      _parseDataUsagePurposeBinaryString (package:iabtcf_consent_info/iabtcf_consent_info.dart:291:10)
I/flutter (17455): #9      _ValueExt.let (package:iabtcf_consent_info/iabtcf_consent_info.dart:345:35)
I/flutter (17455): #10     parseRawConsentInfo.parseDataUsagePurposes (package:iabtcf_consent_info/iabtcf_consent_info.dart:255:26)
I/flutter (17455): #11     parseRawConsentInfo (package:iabtcf_consent_info/iabtcf_consent_info.dart:263:22)
I/flutter (17455): #12     _Ma

**Comment by PasqualePuzio** [comment]
We have the same issue. However there's already a pull request waiting to be merged https://github.com/blaugold/iabtcf_consent_info/pull/12

**Comment by etmyhome** [comment]
@blaugold  any update. ? still range error

**Comment by large** [comment]
Thank you for the push, I cannot wait for a "dead" project, so I am borrowing your version :)
FYI, if you like to have this fix, add this to your pubspec.yaml file:
```dart
  #iabtcf_consent_info: ^3.0.0 #org version
  iabtcf_consent_info: #purpose11 version, fixed arrayfix temp
    git:
      url: https://github.com/bcyran/iabtcf_consent_info.git
      ref: purpose11
      path: packages/iabtcf_consent_info
```

**Comment by blaugold** [comment]
Closed by #12

**Comment by blaugold** [comment]
The fix is available in `iabtcf_consent_info`: ^3.1.0
